### PR TITLE
ORV2-1937 Enable whitelist and forbidNonWhitelistValues

### PR DESCRIPTION
**Pull request by praju-aot**
# Description

Fixes # (issue)
https://jira.th.gov.bc.ca/browse/ORV2-1937

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] npm run test
- [x] npm run test:e2e

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1148-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1148-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1148-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1148-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

**Review by krishnan-aot — APPROVED** [review]

File: vehicles/src/modules/permit/permit.service.ts [hunk 1]
```diff
@@ -259,7 +249,7 @@ export class PermitService {
       });
     }
 
-    if (expired?.toLowerCase() === 'true') {
+    if (!getPermitQueryParamsDto.expired) {
       permitsQuery = permitsQuery.andWhere(
         '(permit.permitStatus IN (:...expiredStatus)OR(permit.permitStatus = :activeStatus AND permitData.expiryDate < :expiryDate))',
```
Comment (zgong-gov) [inline vehicles/src/modules/permit/permit.service.ts]: Is there supposed to be a space between the 'OR' and the two brackets ) and (

File: vehicles/src/modules/permit/dto/request/void-permit.dto.ts [hunk 2]
```diff
@@ -21,18 +22,20 @@ export class VoidPermitDto {
     enum: ApplicationStatus,
     description: 'Revoke or void status for permit.',
     example: ApplicationStatus.REVOKED,
-    required: false,
-  })
+  })  
+  @IsEnum(ApplicationStatus)
```
Comment (zgong-gov) [inline vehicles/src/modules/permit/dto/request/void-permit.dto.ts:26]: There's no change needed on the FE for this right? This field still represents whether or not to void or revoke?

File: vehicles/src/modules/permit/dto/request/create-application.dto.ts [hunk 3]
```diff
@@ -22,78 +31,107 @@ export class CreateApplicationDto {
     example: '',
     required: false,
   })
-  permitId: string;
+  @IsOptional()
+  @IsNumberString()
+  @MaxLength(20)
+  permitId?: string;
```
Comment (zgong-gov) [inline vehicles/src/modules/permit/dto/request/create-application.dto.ts:37]: So most of these changes to the dto (and other dtos) are just making some required fields optional right? I'm guessing there shouldn't be significant changes to the FE corresponding fields? or are there implications that the FE should be aware of?

File: vehicles/src/modules/permit/permit.service.ts (hunk 1 above)
Comment (praju-aot) [reply vehicles/src/modules/permit/permit.service.ts]: yes, however, typeorm and mssql drivers were somehow intelligent enough to resolve it at runtime. I will fix it anyway.

The PR has been tagged 'Don't Review', 'Don't Merge'

File: vehicles/src/modules/permit/dto/request/create-application.dto.ts (hunk 3 above)
Comment (gchauhan-aot) [reply vehicles/src/modules/permit/dto/request/create-application.dto.ts:37]: Yes, no changes to FE. We have updated dtos to place constraints wherever necessary.

File: vehicles/src/modules/permit/dto/request/void-permit.dto.ts (hunk 2 above)
Comment (gchauhan-aot) [reply vehicles/src/modules/permit/dto/request/void-permit.dto.ts:26]: Yes. no changes to FE. Yes this field still represents whether or not to void or revoke.

File: vehicles/src/modules/permit/dto/request/create-application.dto.ts (hunk 3 above)
Comment (zgong-gov) [reply vehicles/src/modules/permit/dto/request/create-application.dto.ts:37]: ok cool, thanks

File: vehicles/src/modules/permit/permit.service.ts (hunk 1 above)
Comment (zgong-gov) [reply vehicles/src/modules/permit/permit.service.ts]: ok I see, np

File: vehicles/src/modules/permit/dto/request/void-permit.dto.ts (hunk 2 above)
Comment (zgong-gov) [reply vehicles/src/modules/permit/dto/request/void-permit.dto.ts:26]: Ok got it, thanks

**Review by zgong-gov — APPROVED** [review]

**Pull request closed without merging**